crowd_anki/github/github_importer.py

```diff
@@ -33,22 +34,38 @@ def import_from_github(self):
                                         'Path:', text='<name>/<repository>')
         if repo and ok:
             self.download_and_import(repo)
-
-    def download_and_import(self, repo):
+    def download_and_import_git(self, github_repo, deck_base_name):
+        repo = None
         try:
-            response = urllib2.urlopen(GITHUB_LINK.format(repo))
-            response_sio = StringIO.StringIO(response.read())
-            with zipfile.ZipFile(response_sio) as repo_zip:
-                repo_zip.extractall(tempfile.tempdir)
+            repo = Repo(Path(self.collection.media.dir()).joinpath("..", "CrowdAnkiGit", deck_base_name, "git"))
+            repo.remote("origin").pull()
+        except git.exc.InvalidGitRepositoryError: # New repository
+            repo = Repo.clone_from(GITHUB_REPO.format(github_repo), Path(self.collection.media.dir()).joinpath("..", "CrowdAnkiGit", deck_base_name, "git"))
+        AnkiJsonImporter.import_deck(self.collection, Path(repo.working_tree_dir))
 
-            deck_base_name = repo.split("/")[-1]
-            deck_directory_wb = Path(tempfile.tempdir).joinpath(deck_base_name + "-" + BRANCH_NAME)
-            deck_directory = Path(tempfile.tempdir).joinpath(deck_base_name)
-            utils.fs_remove(deck_directory)
-            deck_directory_wb.rename(deck_directory)
-            # Todo progressbar on download
-
-            AnkiJsonImporter.import_deck(self.collection, deck_directory)
+    def download_and_import_zip(self, github_repo, deck_base_name):
+        response = urllib2.urlopen(GITHUB_LINK.format(github_repo))
+        response_sio = StringIO.StringIO(response.read())
+        with zipfile.ZipFile(response_sio) as repo_zip:
+            repo_zip.extractall(tempfile.tempdir)
 
+        deck_directory_wb = Path(tempfile.tempdir).joinpath(deck_base_name + "-" + BRANCH_NAME)
+        deck_directory = Path(tempfile.tempdir).joinpath(deck_base_name)
+        utils.fs_remove(deck_directory)
+        deck_directory_wb.rename(deck_directory)
+        # Todo progressbar on download
+        
+        AnkiJsonImporter.import_deck(self.collection, deck_directory)
+    def download_and_import(self, github_repo):
```

Comment:
space before functions. 
as a general level thing - I advise you to install PyCharm (or other IDE of your preference, but I prefer PyCharm) and follow it's style recommendations.